providers/apache/spark/src/airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -279,14 +279,14 @@ def _resolve_connection(self) -> dict[str, Any]:
             if not self.spark_binary:
                 self.spark_binary = extra.get("spark-binary", DEFAULT_SPARK_BINARY)
                 if self.spark_binary is not None and self.spark_binary not in ALLOWED_SPARK_BINARIES:
-                    raise RuntimeError(
+                    raise ValueError(
```

Comment:
This short of change should probably be done separately  -- it's not a good idea to mix provider runtime code changes along with ones that touch core and sdk.